webclient/api/src/main/java/io/helidon/webclient/api/ClientConnection.java

```diff
@@ -61,4 +61,18 @@ public interface ClientConnection extends ReleasableResource {
      * @param readTimeout connection read timeout
      */
     void readTimeout(Duration readTimeout);
+
+    /**
+     * Check whether this connection is allowed to send 100-Continue.
+     *
+     * @return whether 100-Continue is allowed
+     */
+    boolean allowExpectContinue();
```

Comment:
I very much dislike the http specific method on the connection interface, although I understand its purpose.
This could be abstracted away as state with a word (E.g a long) and some mask constant.

Also since this is going to be truly implemented in only one implementation, we should probably use default method. These are public interfaces after all so that would preserve backward compatibility.